packages/core/src/types-hoist/index.ts

```diff
@@ -113,6 +113,7 @@ export type {
   SpanContextData,
   TraceFlag,
 } from './span';
+export type { SpanLink, SpanLinkJSON } from './link';
```

Comment:
m/q: adding this export here, exports the types from core. Do we need need them in other packages? If not, I'd rather not export them until we do (but I might be missing something)